symfony/plugins/orangehrmCorePlugin/Subscriber/ModuleUnderDevelopmentSubscriber.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Core\Subscriber;
+
+use OrangeHRM\Core\Controller\Common\ModuleUnderDevelopmentController;
+use OrangeHRM\Core\Controller\Exception\RequestForwardableException;
+use OrangeHRM\Core\Traits\Service\TextHelperTrait;
+use OrangeHRM\Framework\Event\AbstractEventSubscriber;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class ModuleUnderDevelopmentSubscriber extends AbstractEventSubscriber
+{
+    use TextHelperTrait;
+
+    const MODULE_UNDER_DEVELOPMENT = [
+        'time',
+        'recruitment',
+        'performance',
+        'directory',
+        'dashboard',
+        'maintenance',
+        'buzz'
+    ];
+
+    /**
+     * @inheritDoc
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::REQUEST => [
+                ['onRequestEvent', 100],
+            ],
+        ];
+    }
+
+    public function onRequestEvent(RequestEvent $event)
+    {
+        if ($event->isMasterRequest()) {
+            $modules = self::MODULE_UNDER_DEVELOPMENT; // [indirectly reference the const] looping directly may case run time exception
```

Comment:
are you sure about the comment?

Reply:
not sure, one documentation said like that, hence i followed that way